src/python_testing/TC_IDM_4_2.py

```diff
@@ -0,0 +1,589 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import copy
+import logging
+import random
+import time
+
+import chip.clusters as Clusters
+from chip.ChipDeviceCtrl import ChipDeviceController
+from chip.clusters.Attribute import AttributePath, TypedAttributePath
+from chip.exceptions import ChipStackError
+from chip.interaction_model import Status
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_IDM_4_2(MatterBaseTest):
+
+    async def write_acl(self, acl):
+        result = await self.default_controller.WriteAttribute(self.dut_node_id, [(0, Clusters.AccessControl.Attributes.Acl(acl))])
+        asserts.assert_equal(result[0].Status, Status.Success, "ACL write failed")
+
+    async def get_descriptor_server_list(self, ctrl):
+        return await self.read_single_attribute_check_success(
+            endpoint=0,
+            dev_ctrl=ctrl,
+            cluster=Clusters.Descriptor,
+            attribute=Clusters.Descriptor.Attributes.ServerList
+        )
+
+    async def get_idle_mode_duration(self, ctrl):
+        return await self.read_single_attribute_check_success(
+            endpoint=0,
+            dev_ctrl=ctrl,
+            cluster=Clusters.IcdManagement,
+            attribute=Clusters.IcdManagement.Attributes.IdleModeDuration
+        )
+
+    def verify_attribute_data(self, sub, cluster, attribute, ep=0):
+        sub_attrs = sub
+        if isinstance(sub, Clusters.Attribute.SubscriptionTransaction):
+            sub_attrs = sub.GetAttributes()
+
+        asserts.assert_true(ep in sub_attrs, "Must have read endpoint %s data" % ep)
+        asserts.assert_true(cluster in sub_attrs[0], "Must have read %s cluster data" % cluster.__name__)
+        asserts.assert_true(attribute in sub_attrs[0][cluster],
+                            "Must have read back attribute %s" % attribute.__name__)
+
+    def get_attribute_from_sub_dict(self, sub, cluster, attribute, ep=0):
```

Comment:
why do we have a separate method that replaces one line with another one-line? the method name is quite long as is already so there is likely not even a length of line saving here.

Reply:
Will be removed.